### ISSUE TITLE
False positive unused default when analyzing locals

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#1246]
Repro:

```
$ clojure -M:clj-kondo --config '{:output {:analysis {:locals true}}}' --lint - <<< '(fn [{:keys [a] :or {a 1}}] a)'
<stdin>:1:22: warning: unused default for binding a
``` 

/cc @seancorfield @ericdallo

**Comment by borkdude** [comment]
@ericdallo Fixed on master.

**Comment by ericdallo** [comment]
I think we can close this

**Comment by borkdude** [comment]
@ericdallo I will close it when this is part of the next (non snapshot) release of clj-kondo.
